clients/tfchain-client-go/account.go

```diff
@@ -52,6 +52,14 @@ type AccountInfo struct {
 	Data        Balance   `json:"data"`
 }
 
+type ActivationServiceError struct {
+	Err error
+}
+
+func (e ActivationServiceError) Error() string {
+	return fmt.Sprintf("Activation service error: %s", e.Err.Error())
+}
```

Comment:
IMO defining a custom error type isn't necessary, especially if it doesn't provide extra data in it like status code. An exported generic error type and wrapping it would work fine. Like:

```go
var ErrActivationService = fmt.Errorf("")
...
    return errors.Wrap(ErrActivationService, "")
```

We can check on this error simply with `errors.Is(err, ErrActivationService`

Reply:
Would the error chain in your code be reversed?

Regardless, I modified the structure to include the status code so that you can identify retriable errors if needed. This initially was my plan.